### PR TITLE
Newsletters: add 174 (2021-11-10)

### PR DESCRIPTION
**Pull request by harding**
- [x] Bitcoin Core PR Review Club @glozow
- [x] `Rust-Lightning #1078` Implement channel_type negotiation @adamjonas
- [x] `Rust-Lightning #1144` Penalize failed channels @dongcarl
- [x] `BIPs #1215` Merge pull request #1215 from JeremyRubin/ctv-updates @jnewbery
- [x] Topics/releases @harding

### CODE REVIEW

File: _includes/specials/taproot/en/21-thanks.md [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+Kanon (decryp2kanon),
+kiminuo,
+Luke Dashjr (luke-jr),
+Marco Falke (MarcoFalke),
+Martin Habovštiak (Kixunil),
+Matthew Zipkin (pinheadmz),
+Max Hillebrand (MaxHillebrand),
+Michael Folkson (michaelfolkson),
+Michael Ford (fanquake),
+nopara73,
```
Comment (jnewbery) [inline _includes/specials/taproot/en/21-thanks.md:240]:
```suggestion
Adam Ficsor (nopara73),
```

(from https://youtu.be/QiySI4-MWww?t=11, linked from https://gist.github.com/nopara73)

File: _includes/specials/taproot/en/21-thanks.md [hunk 2]
```diff
@@ -0,0 +1,509 @@
...
+_6102bitcoin,
+AaronvanW,
+achow101,
+aj,
+alec,
+Alexandre_Chery,
+Alistair_Mann,
+amiti,
+_andrewtoth,
+andrewtoth,
```
Comment (jnewbery) [inline _includes/specials/taproot/en/21-thanks.md:272]: Duplicates

File: _includes/specials/taproot/en/21-thanks.md [hunk 3]
```diff
@@ -0,0 +1,509 @@
...
+andytoshi,
+AnthonyRonning,
+ariel25,
+arturogoosnargh,
+AsILayHodling,
+averagepleb,
+bcman,
+belcher,
+benthecarman,
+benthecarman81,
```
Comment (jnewbery) [inline _includes/specials/taproot/en/21-thanks.md:282]: Duplicates

File: _includes/specials/taproot/en/21-thanks.md [hunk 4]
```diff
@@ -0,0 +1,509 @@
...
+entropy5000,
+eoin,
+epson121,
+erijon,
+eris,
+evankaloudis,
+faketoshi,
+fanquake,
+fedorafan,
+fedorafansuper,
```
Comment (jnewbery) [inline _includes/specials/taproot/en/21-thanks.md:332]: Probably duplicates

File: _includes/specials/taproot/en/21-thanks.md [hunk 5]
```diff
@@ -0,0 +1,509 @@
...
+ygrtiugf,
+Yoghurt11411,
+zmnscpxj,
+and
+zndtoshi.
+</i>
+
+**Miner signaling**
+
+We also thank all of the miners since block FIXME that have signaled
```
Comment (jnewbery) [inline _includes/specials/taproot/en/21-thanks.md:485]: Taproot's speedy trial nStartTime was April 24th:

```
        consensus.vDeployments[Consensus::DEPLOYMENT_TAPROOT].nStartTime = 1619222400; // April 24th, 2021
```

The first block for which the activation was STARTED state was therefore the next retarget block [681408](https://blockstream.info/block/000000000000000000020210114d7faf44223e1d620991e468af964514ff96e3?expand), mined on May 1st.

**Review by jnewbery — COMMENTED** [review]
Looks good so far. I've added my write-up for BIPs 1215.

File: _posts/en/newsletters/2021-11-10-newsletter.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+      discrepancies between peers.
+
+    - *Multi-block filters:* these could further reduce the data
+      lightweight clients will need to download.
+
+    - *Committed block filters:* requiring miners commit to the filter
+      for their blocks, reducing the amount of data lightweight clients
+      need to download to monitor for discrepancies between the filters
+      being served by different peers.
+
```
[the remark this answers is not in the document]
Comment (harding) [reply _posts/en/newsletters/2021-11-10-newsletter.md:66]: How do you feel about that?  My personal feeling on the idea is "meh" and that's made me also meh on including it.  If you have any feeling for including it at all, I'm happy to write it up.

[the remark this answers is not in the document]
Comment (jnewbery) [reply _posts/en/newsletters/2021-11-10-newsletter.md:66]: I'm neutral. I haven't thought about it enough to form an opinion.

The fact that Laolu left it out of the original email does suggest that it's a bit of an afterthought, so I don't have any problem with leaving it out of the newsletter.

File: _posts/en/newsletters/2021-11-10-newsletter.md [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+  4. Clients such as RPC (e.g. `sendrawtransaction`) and the wallet (e.g.
+     `sendtoaddress`) submit their transactions to the node using
+     [`BroadcastTransaction()`][atmp client], which calls ATMP. The
+     `testmempoolaccept` RPC also calls ATMP with `test_accept` set to `true`. These
+     are examples of calls from components 'external' to validation."
+  a1link="https://bitcoincore.reviews/23173#l-80"
+
+  q2="What does `CTxMemPool::check()` do? Whose responsibility is it to call
+    that function?"
+  a2="`CTxMemPool::check()` checks that each transaction's inputs correspond to
```
Comment (jnewbery) [inline _posts/en/newsletters/2021-11-10-newsletter.md:107]: I don't think this makes it clear that that this is checking the entire contents of the mempool (and could instead be checking transactions that are being submitted to the mempool). How about something like:

```suggestion
  a2="`CTxMemPool::check()` does checks on _all_ the transactions currently in the mempool. It checks that each transaction's inputs correspond to
```

File: _posts/en/newsletters/2021-11-10-newsletter.md [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+net\_processing, and multiple contributors wish to minimize the usage of`cs_main`.
+The variable was named when validation functionality was housed in a main.cpp
+file. The prefix, `cs`, stands for critical section."
+a0link="https://bitcoincore.reviews/23173#l-45"
+
+  q1="Which components currently call `AcceptToMemoryPool`? Which of the ATMP calls are from external client code and which are from inside validation?"
+  a1="Excluding calls from tests, there are four call sites:
+  1. When the node starts, it [loads][atmp disk] transactions from mempool.dat and
+     calls ATMP to re-validate the transactions and restore mempool contents. This
+     call originates from a component 'external' to validation.
```
Comment (jnewbery) [inline _posts/en/newsletters/2021-11-10-newsletter.md:92]: I disagree that this is external to validation. `LoadMempool()` is contained within validation.cpp and is called from a `CChainState` object (see https://bitcoincore.reviews/23173#l-107).

File: _posts/en/newsletters/2021-11-10-newsletter.md [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+  a2link="https://bitcoincore.reviews/23173#l-122"
+
+  q3="What does the `bypass_limits` argument do? In which circumstances is ATMP
+    called with it set to true?"
+  a3="When `bypass_limits` is true, the mempool maximum size and minimum feerate
+  are not enforced. For example, if the mempool is full and its dynamic mempool
+  minimum feerate is 3sat/vB, an individual transaction with a 1sat/vB feerate may
+  be accepted. ATMP is called with `bypass_limits` during a [reorg][atmp bypass
+  limits]; these transactions may have low individual feerates but high descendant
+  feerates. The size of the transactions is limited to
```
Comment (jnewbery) [inline _posts/en/newsletters/2021-11-10-newsletter.md:124]:
```suggestion
  feerates. The total size of all transactions to re-add to the mempool during a reorg is limited to
```

File: _posts/en/newsletters/2021-11-10-newsletter.md [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+repo], [Hardware Wallet Interface (HWI)][hwi repo],
+[Rust Bitcoin][rust bitcoin repo], [BTCPay Server][btcpay server repo],
+[BDK][bdk repo], [Bitcoin Improvement Proposals (BIPs)][bips repo], and
+[Lightning BOLTs][bolts repo].*
+
+- [Rust-Lightning #1078][] adds `channel_type` negotiation as defined in [BOLTs
+  #880][] and covered in [Newsletter #165][news165 bolts-880]. This
+  implementation does not currently send the proposed feature bit from [BOLTs
+  #906][], which are likely to be required for turbo channels, but if not, will
+  be needed for anchor channels.
```
Comment (jnewbery) [inline _posts/en/newsletters/2021-11-10-newsletter.md:180]: BOLTs 906 isn't required for turbo channels/anchor outputs. When Matt says "It is likely to be required for turbo channels [...]", he's referring to support for BOLTs 880.

```suggestion
  #906][]. [BOLTs #880][] is required for [anchor channels][topic anchor outputs] and may also be needed to support [turbo channels](https://bitcoinops.org/en/newsletters/2021/07/07/#zero-conf-channel-opens).
```

File: _posts/en/newsletters/2021-11-10-newsletter.md [hunk 11]
```diff
@@ -0,0 +1,225 @@
...
+
+- [Rust-Lightning #1144][] adds a penalization mechanism to the route scoring
+  logic. The penalty is applied to failed channels between payment retries to
+  inform the path finding algorithm of potentially faulty channels.
```
Comment (jnewbery) [inline _posts/en/newsletters/2021-11-10-newsletter.md:184]: "pathfinding" is slightly more common in our repo than "path finding". I suggest that we standardize to pathfinding and add it to the style guide.

File: _posts/en/newsletters/2021-11-10-newsletter.md [hunk 12]
```diff
@@ -0,0 +1,225 @@
...
+John Newbery to add a new `ChainstateManager::ProcessTransaction()` interface
+function responsible for processing transactions as candidates to the mempool
+and performing mempool consistency checks. The review club discussed the
+current interface for adding transactions to the mempool.
+
+{% include functions/details-list.md
+  q0="What is `cs_main`? Why is it called `cs_main`?"
+  a0="`cs_main` is a mutex intended to synchronize multi-threaded access to
+validation state. In reality, it also guards non-validation data in
+net\_processing, and multiple contributors wish to minimize the usage of`cs_main`.
```
[the remark this answers is not in the document]
Comment (glozow) [reply _posts/en/newsletters/2021-11-10-newsletter.md:83]: not sure, my typical style is to keep file names in plaintext, but i'm happy to go with whatever the convention is in bitcoinops?

File: _posts/en/newsletters/2021-11-10-newsletter.md (hunk 8 above)
Comment (glozow) [reply _posts/en/newsletters/2021-11-10-newsletter.md]: I think this was a typo - thanks!

File: _posts/en/newsletters/2021-11-10-newsletter.md (hunk 7 above)
Comment (glozow) [reply _posts/en/newsletters/2021-11-10-newsletter.md]: thanks, changed to "checks that all transactions' inputs correspond to available UTXOs and performs internal consistency checks over the entire mempool."

File: _posts/en/newsletters/2021-11-10-newsletter.md (hunk 9 above)
Comment (glozow) [reply _posts/en/newsletters/2021-11-10-newsletter.md]: fixed, thanks

File: _posts/en/newsletters/2021-11-10-newsletter.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (glozow) [reply _posts/en/newsletters/2021-11-10-newsletter.md]: I changed this to "data used in P2P logic"

File: _includes/specials/taproot/en/21-thanks.md [hunk 13]
```diff
@@ -0,0 +1,509 @@
...
+<i>
+Adam Back,
+Andrea Barontini,
+Andreas Schildbach,
+Andrew Chow,
+Andrew Poelstra,
+Anthony Towns,
+Antoine Riard,
+Ariel Lorenzo-Luaces,
+Ariel Luaces,
```
[the remark this answers is not in the document]
Comment (harding) [reply _includes/specials/taproot/en/21-thanks.md:31]: Confirmed, they always sign their name in the email body with the hyphenated form.

**Review by Xekyo — COMMENTED** [review]
I'd like to propose that we generally separate value and unit of measurement by a (protected) space as per the SI standard: https://en.wikipedia.org/wiki/Space_(punctuation)#Unit_symbols_and_numbers

**Comment by harding** [comment]
Pushed edits (and updates to the Style guide) for all feedback (I hope).  Thank you everyone for the reviews and suggestions!  Also reviewed @glozow @dongcarl @adamjonas @jnewbery contributions (thank you!) with a few tiny edits.

Updated releases/RCs and added topic links.

I'll try to check again tonight before bed for any final feedback, thank you everyone!

**Comment by bitschmidty** [comment]
Looks great team. Thank you @harding @dongcarl @jnewbery @adamjonas and @glozow for authoring and @Xekyo for the 👓

**Pull request merged**